src/tests/conftest.py

```
@@ -24,7 +21,9 @@ async def reset_db():
 @pytest_asyncio.fixture(scope="function")
 async def client():
     """Provide an asynchronous test client for making HTTP requests."""
-    async with AsyncClient(transport=ASGITransport(app=app), base_url="http://test") as async_client:
+    async with AsyncClient(
+        transport=ASGITransport(app=app), base_url="http://test"
```

Async test client: using `AsyncClient` with `ASGITransport(app=app)` and base_url='http://test' is fine for the tests. Consider adding a small default timeout (e.g. `timeout=5.0`) to the `AsyncClient` to fail fast on hung requests during CI. The current context manager usage correctly yields the client and ensures cleanup.